plugins/pulp_rpm/plugins/importers/yum/listener.py

```diff
@@ -220,3 +221,11 @@ def download_succeeded(self, report):
             if not added_unit.downloaded:
                 added_unit.downloaded = True
```

Comment:
@goosemania do you think we could add lines #L227-L230 right after we set the downloaded flag to true and before saving the unit?

Reply:
I don't think so. Is there a situation when unit is already marked as downloaded but modular flag is not set yet? If yes (I think this is the case), then they should stay separate.
If no, then it's strange that we check if downloaded flag is set or not and we don't just blindly set it every time.

Reply:
i probably picked up not the best line where to move the code, but the intention was to minimize the number of the save call. that's just a suggestion i am ok with current approach as well.
once we can access headers at ( `headers = rpm_parse.package_headers(report.destination)` ) the means the unit is downloaded and we could flip the modular flag earlier